src/pal/pal_sk_posix.c

```diff
@@ -1389,7 +1389,9 @@ void pal_socket_free(
     mem_free_type(pal_socket_t, sock);
 }
 
+#ifndef USE_CTEST // avoid conflicting definition of clockid_t in os_mock.h:882 and /usr/include/time.h
```

Comment:
I would rather like to fix this in os_mock.h then if we absolutely cannot avoid including time.h at all e.g. because it comes through standard headers on the distro.  Then I would also try to find the actual version of the header or some other identifying system info to preprocess the clock_id type out of os_mock.h.  If possible.